roscala/src/main/scala/coop/rchain/rosette/prim/queue.scala

```diff
@@ -0,0 +1,203 @@
+package coop.rchain.rosette.prim
+
+import coop.rchain.rosette.Ob.ABSENT
+import coop.rchain.rosette.{Ctxt, Fixnum, Ob, PrimErrorWrapper, Queue, RblBool, Tuple}
+import coop.rchain.rosette.macros.{checkArgumentMismatch, checkTypeMismatch}
+import coop.rchain.rosette.prim.Prim._
+
+object queue {
+
+  object queueNew extends Prim {
+    override val name: String = "queue-new"
+    override val minArgs: Int = 0
+    override val maxArgs: Int = 0
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Queue] =
+      Right(Queue.create())
+  }
+
+  object queueDepth extends Prim {
+    override val name: String = "queue-depth"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Queue]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val q = ctxt.argvec.elem.head.asInstanceOf[Queue]
+      Right(q.nElems.asInstanceOf[Fixnum])
+    }
+  }
+
+  object queueIsEmpty extends Prim {
+    override val name: String = "queue-empty?"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Queue]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblBool] = {
+      val q = ctxt.argvec.elem.head.asInstanceOf[Queue]
+      Right(RblBool(q.isEmpty()))
+    }
+  }
+
+  object queueEnqueue extends Prim {
+    override val name: String = "queue-enqueue"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Queue] =
+      ctxt.argvec.elem.head match {
+        case q: Queue =>
+          val ob = ctxt.argvec.elem(1)
+          Right(q.enqueue(ob))
+        case _ =>
+          Left(ArgumentMismatch("The first element should be a queue"))
+      }
+  }
+
+  object queueDeQueue extends Prim {
+    override val name: String = "queue-dequeue"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Ob] =
+      ctxt.argvec.elem.head match {
+        case q: Queue =>
+          if (q.isEmpty()) {
+            Right(ABSENT)
```

Comment:
I know it's crazy, but the primitives are not consistent in their handling of error conditions.

queue-dequeue returns PrimError on an empty queue. queue-pat-dequeue returns #absent on an empty queue.

Please compare all of the primitives with the C++ code and return the appropriate thing for each error condition.